pkg/controller/worker/machines_test.go

```diff
@@ -553,13 +555,21 @@ var _ = Describe("Machines", func() {
 					Expect(err).NotTo(HaveOccurred())
 
 					// Test workerDelegate.GenerateMachineDeployments()
-
 					result, err := workerDelegate.GenerateMachineDeployments(context.TODO())
 					Expect(err).NotTo(HaveOccurred())
 					Expect(result).To(Equal(machineDeployments))
 				})
 			})
 
+			It("should return error when failing to delete AlicloudMachineClass", func() {
```

Comment:
Would be good to cover more cases if you could replicate these test cases [here]( https://github.com/gardener/gardener-extension-provider-aws/pull/212/files#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R640-R678). Would be better to catch future bugs/regressions (if any).

Reply:
I added one more test case of failing to decode spec.status, see https://github.com/gardener/gardener-extension-provider-alicloud/pull/224/files#diff-12a1fa4ba4f4250731a53989c476c9407e832ccecae17ad3fc1b2eee100ad7c3R563-R570

Reply:
because this part of code (https://github.com/gardener/gardener-extension-provider-alicloud/blob/0eb323467cb87aa25c64a83255915cc8cecf2ddc/pkg/controller/worker/machines_test.go#L511-L526) includes the logic that successful to delete all old `AlicloudMachineClass`, so I didn't add the test case of deleting all old AlicloudMachineClass.